cherno/guider.py

```diff
@@ -192,171 +200,127 @@ def __init__(
         self.target_rms = target_rms
         self.pids = AxesPID(command.actor if command is not None else None)
 
+        # reference to solve pointing instance, initialize to None
+        self.solve_pointing = None
+
         # To cache Gaia sources.
         self._gaia_sources: dict = {}
 
         # Clear RMS history.
         if self.command.actor:
             self.command.actor.state.rms_history.clear()
 
+        # setup task queue for writing proc-*.fits files
+        self.write_background_tasks = set()
+        self.write_executor = ProcessPoolExecutor(
+            max_workers=3,
+            mp_context=mp.get_context("spawn"),
+        )
+
         self._database_lock = asyncio.Lock()
 
     def set_command(self, command: ChernoCommandType):
         """Sets the command."""
 
         self.command = command
 
-    async def process(
+    def check_convergence(
         self,
-        command: ChernoCommandType | None,
-        images: list[PathLike],
-        write_proc: bool = True,
-        overwrite: bool = False,
-        correct: bool = True,
-        full_correction: bool = False,
-        offset: list[float] | None = None,
-        scale_rms: bool = True,
-        wait_for_correction: bool = True,
-        only_radec: bool = False,
-        auto_radec_min: int = 2,
-        use_astrometry_net: bool | None = None,
-        use_gaia: bool = True,
-        gaia_phot_g_mean_mag_max: float | None = None,
-        gaia_cross_correlation_blur: float | None = None,
-        fit_all_detections: bool = True,
-        fit_focus: bool = True,
-        plot: bool | None = None,
-        stop_at_target_rms: bool = False,
+        ex_data: list[ExtractionData],
+        offset: list[float] | None,
     ):
-        """Performs extraction and astrometry."""
+        if self.solve_pointing is None:
+            return False
 
-        if command is not None:
-            self.set_command(command)
+        nSources = numpy.sum([len(ex.regions) for ex in ex_data])
+        if nSources < 6:
+            return False
 
-        self.command.info("Extracting sources.")
+        ex = ex_data[0]
 
-        ext_data = await asyncio.gather(
-            *[
-                run_in_executor(
-                    self.extractor.process,
-                    im,
-                    plot=plot,
-                    executor="process",
-                )
-                for im in images
-            ]
-        )
+        if ex.exposure_no != self.solve_pointing.imgNum + 1:
```

Comment:
This would fail if `imgNum` is `None`, which is the default value. I think you can add a 

```
if self.solve_pointing.imgNum is None:
    return False
```

Reply:
I think this is ok, because earlier in check_convergence there is a check for self.solve_pointing != None.  If it passes that check then solve_pointing has an imgNum attr.